libparsec/crates/client/src/certif/list.rs

```diff
@@ -151,6 +131,90 @@ pub(super) async fn list_users(
         .await?
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum CertifGetUserInfoError {
+    #[error("No user with this UserID")]
+    NonExisting,
+    #[error("Component has stopped")]
+    Stopped,
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<CertifStoreError> for CertifGetUserInfoError {
+    fn from(value: CertifStoreError) -> Self {
+        match value {
+            CertifStoreError::Stopped => Self::Stopped,
+            CertifStoreError::Internal(err) => err.into(),
+        }
+    }
+}
+
+pub(super) async fn get_user_info_for_user_certificates(
+    user_certificate: std::sync::Arc<UserCertificate>,
```

Comment:
In theory it cleaner to use references here:
```suggestion
    user_certificate: &UserCertificate,
    revoked_certificate: Option<&RevokedUserCertificate>,
```

And if I recall correctly you can turn a `Option<Arc<X>>` into a `Option<&X>` in the caller just by doing `x.as_ref()

However it is an internal helper, so if the caller becomes too cumbersome we should just stick to the simpler approach :+1:

Reply:
`x.as_ref()` returns `Option<&Arc<X>>` (not `Option<&X>`). 

The only way I found is to take a ref to the Option and map `Option<Arc<X>>` to `Option<&X>` with `x.as_ref().map(|x| x.as_ref())`

So caller changes from:
```rust
cook_user_info_from_certificates(user_certificate, revoked_certificate, update_certificate)
```
to
```rust
cook_user_info_from_certificates(
    user_certificate.as_ref(),
    revoked_certificate.as_ref().map(|x| x.as_ref()),
    update_certificate.as_ref().map(|x| x.as_ref()),
)
```
Any preference?

Reply:
@mmmarcos What you are looking is for `as_deref`

Reply:
Done, thx @FirelightFlagboy!